app/models/spree/package_builder.rb

```diff
@@ -106,7 +106,7 @@ def packages(package)
       end
 
       item_specific_packages.each do |package|
-        packages << ::ActiveShipping::Package.new(package.at(0), [package.at(1), package.at(2), package.at(3)], units: :imperial)
+        packages << ::ActiveShipping::Package.new(package.weight * multiplier, [package.length, package.width, package.height], units: units)
```

Comment:
Everywhere else in the code, the unit is pulled from the config so I'd assume people who wants to use :imperial are setting the value themselves otherwise you'd get weird results

I'd also expect the behavior to be different because this whole part looks bugged to me considering that product with product packages are double-packed (once based on weight and once based on product_packages)